### ISSUE TITLE
Question: is there a warmup time?

### ISSUE THREAD

**Issue by AshleyColemanBigHealth** [archriss/react-native-snap-carousel#483]
### question: sometimes `.snapToItem(index)` fails

I call `this.carousel.snapToItem(index)` within `componentDidUpdate()`. This sometimes works, but sometimes doesn't (to "fix" this I put the call in the `setTimeout`).

I was just wondering if there was a "warm up" time for the carousel (or the underlying Views).

And if so, how can I check that it is "ready".

Thanks

**Comment by JonghunYu** [comment]
I have same issue and it turns out that `renderItem` function in carousel is excuted after `componentDidMount` of parent component is excuted. I guess same thing would happen for the `componentDidMount`. I couldn't figure out how to avoid this issue except `setTimeout` yet

**Comment by baeteja** [comment]
I have the same issue, is there no better way around this?

**Comment by wsliaw** [comment]
you should try using `firstItem` props:
```
<Carousel
  firstItem={2}
/>
```

**Comment by fedu44** [comment]
Yeah, I have the same issue. I dont really know why that is happening. Prob the is a race condition that make that strange behaviour.

**Comment by bleudreams** [comment]
I am also running into this problem. Has anyone managed to make progress?

**Comment by r-bman** [comment]
Struggling with this too, any help would be very much appreciated.

**Comment by dohooo** [comment]
Sorry, please allow me to advertise for my open source library! ~
I think this library [react-native-reanimated-carousel](https://github.com/zwh1666258377/react-native-reanimated-carousel) will solve your problem. It is a high performance and very simple component, complete with React-Native reanimated 2
